File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/implement/StatisticConvertImp.java
package com.platdmit.simplecloudmanager.domain.converters.implement;

import com.platdmit.simplecloudmanager.data.api.models.ApiStatistic;
import com.platdmit.simplecloudmanager.data.database.entity.DbStatistic;
import com.platdmit.simplecloudmanager.domain.converters.StatisticConverter;
import com.platdmit.simplecloudmanager.domain.models.Statistic;

import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

import java.util.ArrayList;
import java.util.List;

public class StatisticConvertImp implements StatisticConverter{

    @Override
    public DbStatistic fromApiToDb(ApiStatistic apiStatistic, long serverId) {
        return new DbStatistic(
                serverId,
                timeConvert(apiStatistic.getTime()),
                Float.parseFloat(apiStatistic.getRam()),
                Float.parseFloat(apiStatistic.getCpu())
        );
    }

    @Override
    public Statistic fromDbToDomain(DbStatistic dbStatistic) {
        return new Statistic(
                dbStatistic.getTime(),
                dbStatistic.getRamVal(),
                dbStatistic.getCpuVal()
        );
    }

    @Override
    public List<DbStatistic> fromApiToDbList(List<ApiStatistic> apiStatistics, long serverId) {
        List<DbStatistic> convertList = new ArrayList<>();
        for (ApiStatistic apiStatistic : apiStatistics) {
            try {
                convertList.add(fromApiToDb(apiStatistic, serverId));
            } catch (NullPointerException ignored){}
        }
        return convertList;
    }

    @Override
    public List<Statistic> fromDbToDomainList(List<DbStatistic> dbStatistics) {
        List<Statistic> convertList = new ArrayList<>();
        for (DbStatistic dbStatistic : dbStatistics) {
            try {
                convertList.add(fromDbToDomain(dbStatistic));
            } catch (NullPointerException ignored){}
        }
        return convertList;
    }

    private String timeConvert(String date){
        try {
            DateTimeFormatter dateTimeFormatter = DateTimeFormat.forPattern("DD.MM.yyyy HH:mm:SS");
            return DateTimeFormat.forPattern("HH:mm").print(dateTimeFormatter.parseDateTime(date));
        } catch (NullPointerException | IllegalArgumentException ignored){
            return "";
        }
    }
}

File: domain/src/main/java/com/platdmit/domain/converters/implement/StatisticConvertImp.kt
package com.platdmit.domain.converters.implement

import com.platdmit.data.api.models.ApiStatistic
import com.platdmit.data.database.entity.DbStatistic
import com.platdmit.domain.converters.StatisticConverter
import com.platdmit.domain.models.Statistic
import org.joda.time.format.DateTimeFormat
import java.util.*

class StatisticConvertImp : StatisticConverter {
    override fun fromApiToDb(apiStatistic: ApiStatistic, serverId: Long): DbStatistic {
        return DbStatistic(
                serverId,
                timeConvert(apiStatistic.time), apiStatistic.ram.toFloat(), apiStatistic.cpu.toFloat())
    }

    override fun fromDbToDomain(dbStatistic: DbStatistic): Statistic {
        return Statistic(
                dbStatistic.time,
                dbStatistic.ramVal,
                dbStatistic.cpuVal
        )
    }

    override fun fromApiToDbList(apiStatistics: List<ApiStatistic>, serverId: Long): List<DbStatistic> = apiStatistics.map { fromApiToDb(it, serverId) }

    override fun fromDbToDomainList(dbStatistics: List<DbStatistic>): List<Statistic> = dbStatistics.map { fromDbToDomain(it) }

    private fun timeConvert(date: String): String {
        return try {
            val dateTimeFormatter = DateTimeFormat.forPattern("DD.MM.yyyy HH:mm:SS")
            DateTimeFormat.forPattern("HH:mm").print(dateTimeFormatter.parseDateTime(date))
        } catch (ignored: NullPointerException) {
            ""
        } catch (ignored: IllegalArgumentException) {
            ""
        }
    }
}